fsspec/implementations/cache_metadata.py

```diff
@@ -41,6 +49,28 @@ def __init__(self, storage: list[str]):
         self._storage = storage
         self.cached_files: list[Detail] = [{}]
 
+        # Private attribute to force saving of metadata in pickle format rather than
+        # JSON for use in tests to confirm can read both pickle and JSON formats.
+        self._force_save_pickle = False
+
+    def _load(self, fn: str) -> Detail:
+        """Low-level function to load metadata from specific file"""
+        try:
+            with open(fn, "r") as f:
+                return json.load(f)
+        except Exception:
```

Comment:
Exception should be way more specific.

Reply:
IOError or JSON decode error. The latter will depend on the JSON library used, I don't think they nicely call it the same thing - but probably both subclass ValueError.

Reply:
The `Exception` was purposely generic. If anything fails with the JSON load it tried the pickle load. A corner case failure in the first will be repeated in the second.

It is not as simple as IOError or JSON decode errror. It is easy to obtain a UnicodeDecodeError for example, and I expect there are others that I haven't explicitly tested for.

But I don't have a strong opinion either way. I have changed it to a `ValueError` as suggested.